stm32h5 device supports GPdma

FRASTM opened the pull request:
Add the support of the GP DMA  to the stm32H5 serie
Two GPDMA instances are defined with 8 channels and 140 request
The type of DMA for the stm32h5 serie is similar to the stm32u5.


Signed-off-by: Francois Ramu [francois.ramu@st.com](mailto:francois.ramu@st.com)

FRASTM committed:
tests: drivers: dma testing on the stm32h573i_dk board

Configure the tests/drivers/dma/loop_transfer
and the tests/drivers/dma//chan_blen_transfer
to run on the stm32h573i_dk : use gpdma1 or gpdma2.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
boards: arm: stm32h5 disco kit stm32h573i_dk enables dma

Add the DMA instance for the stm32h573i_dk disco kit
from STMicroelectronics.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
dts: arm: stm32h5 serie adds nodes for GP DMA

Adds the nodes for the GPDMA 1 & 2  peripherals
to the stm32h5 serie.
Each instance has 8 channels and 140 DMA requests.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
Running the tests/drivers/dma/chan_blen_transfer on the stm32h573i disco:
```
TESTSUITE dma_m2m succeeded                                                           
                                                                                      
------ TESTSUITE SUMMARY START ------                                                 
                                                                                      
SUITE PASS - 100.00% [dma_m2m]: pass = 4, fail = 0, skip = 0, total = 4 duration = 8.s
 - PASS - [dma_m2m.test_test_dma0_m2m_chan0_burst16] duration = 2.014 seconds         
 - PASS - [dma_m2m.test_test_dma0_m2m_chan0_burst8] duration = 2.014 seconds          
 - PASS - [dma_m2m.test_test_dma0_m2m_chan1_burst16] duration = 2.014 seconds         
 - PASS - [dma_m2m.test_test_dma0_m2m_chan1_burst8] duration = 2.014 seconds          
                                                                                      
------ TESTSUITE SUMMARY END ------      
```

erwango commented on drivers/dma/dma_stm32u5.c:
```diff
@@ -24,9 +24,13 @@ LOG_MODULE_REGISTER(dma_stm32, CONFIG_DMA_LOG_LEVEL);
 #define DT_DRV_COMPAT st_stm32u5_dma
 
-/* STM32U5 soc has only one GPDMA instance of 15 channels */
+/* STM32U5 soc has only one GPDMA instance of 16 channels */
```
Question on the comment: Is it the GPDMA IP that can have up to 16 channels, or the U5 IP ?

erwango reviewed (dismissed):
One minor comment.
Not blocking as the code is fixed in next commit (which is ok if we consider 2 commits are dealing with different stuff)

FRASTM commented on drivers/dma/dma_stm32u5.c:
```diff
@@ -24,9 +24,13 @@ LOG_MODULE_REGISTER(dma_stm32, CONFIG_DMA_LOG_LEVEL);
+
+#if DT_NODE_HAS_STATUS(DT_DRV_INST(0), okay)
+#define DMA_STM32_0_STREAM_COUNT DT_INST_PROP(0, dma_channels)
 #endif /* DT_NODE_HAS_STATUS(DT_DRV_INST(0), okay) */
```
yes, removed in the other commit.
2 commits placed on the stm32u5_dma, could still be made of one.

FRASTM committed:
drivers: dma: stm32u5 dma driver with two dma instances

Add a second instance of DMA to the stm32 dma driver from the DTS
That can be found in the stm32H5 serie, where 2 GPDMA with 8 channels
are available. Rebuilding with Macro.
Use the dma_channels property of the dedive tree to count the nb
of dma channels: 16 or 8 (like in stm32h5).

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
Merge two commits for the driver
add the _zephyr/dt-bindings/dma/stm32_dma.h_ to the device DTS
